### ISSUE TITLE
filter_lines failing with Chef::Exceptions::DeprecatedFeatureError

### ISSUE THREAD

**Issue by krusolu** [sous-chefs/line#158]
### Cookbook version
2.3.2

### Chef-client version
13.8.5

### Platform Details
Oracle Linux 6

### Scenario:
Trying to modify a file by adding a line after an existing line

### Steps to Reproduce:
Create file /tmp/mongod.conf - snippet included:
```
# mongod.conf
# for documentation of all options, see:
#   http://docs.mongodb.org/manual/reference/configuration-options/
# where to write logging data.
systemLog:
  destination: file
  logAppend: true
  path: /var/log/mongodb/mongod.log

# Where and how to store data.
storage:
  dbPath: /apps_data_01/mongo
  journal:
    enabled: true
#  engine:
#  mmapv1:
#  wiredTiger:
```
Create a recipe:
```
mongodb_config_file = '/tmp/mongod.conf'

delete_lines "#{mongodb_config_file}: remove smallFiles option" do
  path mongodb_config_file
  pattern 'smallFiles'
end

replace_or_add "#{mongodb_config_file}: uncomment mmapv1" do
  path mongodb_config_file
  pattern 'mmapv1'
  line '  mmapv1:'
end

filter_lines "#{mongodb_config_file}: set smallFiles to true" do
  path mongodb_config_file
  filters after: ['mmapv1', "    smallFiles: true"]
end
```
run recipe

### Expected Result:
Expecting the file to result in:
```
[...]
  mmapv1:
    smallFiles: true
[...]
```

### Actual Result:
```
         * filter_lines[/etc/mongod.conf: set smallFiles to true] action edit[2019-06-25T11:34:56-04:00] ERROR: rename filters to new_resource.filters (CHEF-19)/tmp/kitchen/cache/cookbooks/line/libraries/helper.rb:53:in `filter_method'.
       Please see https://docs.chef.io/deprecations_namespace_collisions.html for further details and information on how to correct this problem. at /opt/chef/embedded/lib/ruby/gems/2.4.0/gems/chef-13.8.5/lib/chef/formatters/doc.rb:419:in `deprecation'


           ================================================================================
           Error executing action `edit` on resource 'filter_lines[/etc/mongod.conf: set smallFiles to true]'
           ================================================================================

           Chef::Exceptions::DeprecatedFeatureError
           ----------------------------------------
           "rename filters to new_resource.filters (CHEF-19)/tmp/kitchen/cache/cookbooks/line/libraries/helper.rb:53:in `filter_method'.\nPlease see https://docs.chef.io/deprecations_namespace_collisions.html for further details and information on how to correct this problem. at /opt/chef/embedded/lib/ruby/gems/2.4.0/gems/chef-13.8.5/lib/chef/formatters/doc.rb:419:in `deprecation'"

           Cookbook Trace:
           ---------------
           /tmp/kitchen/cache/cookbooks/line/libraries/helper.rb:53:in `filter_method'
           /tmp/kitchen/cache/cookbooks/line/libraries/helper.rb:67:in `invoke_filter'
           /tmp/kitchen/cache/cookbooks/line/libraries/helper.rb:30:in `block in apply_filter'
           /tmp/kitchen/cache/cookbooks/line/libraries/helper.rb:29:in `each'
           /tmp/kitchen/cache/cookbooks/line/libraries/helper.rb:29:in `apply_filter'
           /tmp/kitchen/cache/cookbooks/line/resources/filter_lines.rb:60:in `block in class_from_file'

           Resource Declaration:
           ---------------------
           suppressed sensitive resource output

           Compiled Resource:
           ------------------
           suppressed sensitive resource output

           System Info:
           ------------
           chef_version=13.8.5
           platform=oracle
           platform_version=6.9
           ruby=ruby 2.4.3p205 (2017-12-14 revision 61247) [x86_64-linux]
           program_name=chef-solo worker: ppid=2023;start=11:34:42;
           executable=/opt/chef/bin/chef-solo
```

**Comment by JohnRoesler** [comment]
Looks like chef is even telling us how to fix it `rename filters to new_resource.filters` here https://github.com/sous-chefs/line/blob/master/libraries/helper.rb#L53-L54

**Comment by krusolu** [comment]
The chef suggested fix does not work.  I think something needs to fixed in the filter_helper Line::Filter class to implement the calling methods.

**Comment by MarkGibbons** [comment]
Thank you for the report. I'll take a look.

**Comment by MarkGibbons** [comment]
I think the filters method will have to be renamed in the helpers code.  I see the failure on chef 13 and 15.  Works fine on 14.  The fix looks simple. Testing now.

**Comment by MarkGibbons** [comment]
Should be fixed in 2.3.3
